In [1]:
%run ../scripts/notebook_settings.py
import glob

Performing some checks to understand Vasilis metadata.

In [2]:
table_desc = "~/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/plots/SupTable_Sample_Stats_wGT_QC.tsv"

In [3]:
vasili_table = pd.read_csv(table_desc, sep="\t")
vasili_table.columns

Index(['ID', 'ID_PGDP', 'group', 'family', 'genus', 'species', 'subspecies',
       'species_full', 'species_short', 'species_genotyping', 'reference',
       'SEX', 'len_chrA', 'len_chrA_contigs_shorter_than_1Mb', 'len_chrX',
       'len_chrX_contigs_shorter_than_1Mb', 'len_chrY',
       'len_chrY_contigs_shorter_than_1Mb', 'total_seq_chrA', 'total_seq_chrX',
       'total_seq_chrY', 'len_cov_chrA', 'len_cov_chrX', 'len_cov_chrY',
       'frac_cov_chrA', 'frac_cov_chrX', 'frac_cov_chrY', 'frac_cov_gw',
       'rel_frac_cov_YX', 'nonN_frac', 'frac_cov_gw_cor', 'threshold',
       'threshold_chrX', 'good_frac_cov', 'cov_chrA', 'cov_chrX', 'cov_chrY',
       'cov_gw', 'rel_cov_chrX', 'gSEX', 'gSEX_source', 'callable_chrA',
       'callable_chrX', 'GT00_chrA', 'GT01_chrA', 'GT11_chrA', 'GTmis_chrA',
       'GT00_chrX', 'GT0_chrX', 'GT01_chrX', 'GT11_chrX', 'GT1_chrX',
       'GTmis_chrX', 'het_chrA', 'het_chrX', 'rel_chrX_het', 'hom_alt_chrA',
       'hom_alt_chrX', 'hem_alt_chrX', 'rel_c

In [4]:
#vasili_table.loc[(vasili_table.genus == "Papio") & (~vasili_table.relatives.isna())
#                & (vasili_table.ID_PGDP.str.startswith("PD"))][["ID_PGDP", "ID", "species",
#                                                                "relatives", "remove_as_relative",
#                                                                "remove_as_duplicate"]].to_csv("papio_all.csv", index=False)

In [5]:
vasili_table.species_full

0       Allenopithecus_nigroviridis_ssp
1       Allenopithecus_nigroviridis_ssp
2       Allenopithecus_nigroviridis_ssp
3                      Aotus_azarai_ssp
4                      Aotus_azarai_ssp
                     ...               
3235              Varecia_variegata_ssp
3236              Varecia_variegata_ssp
3237              Varecia_variegata_ssp
3238              Varecia_variegata_ssp
3239              Varecia_variegata_ssp
Name: species_full, Length: 3240, dtype: object

In [6]:
vasili_table.reference

0       Allenopithecus_nigroviridis
1       Allenopithecus_nigroviridis
2       Allenopithecus_nigroviridis
3                   Aotus_nancymaae
4                   Aotus_nancymaae
                   ...             
3235              Varecia_variegata
3236              Varecia_variegata
3237              Varecia_variegata
3238              Varecia_variegata
3239              Varecia_variegata
Name: reference, Length: 3240, dtype: object

In [7]:
len(vasili_table), vasili_table.gSEX_source.unique()

(3240, array(['chrX', 'metadata', 'default', 'chrY'], dtype=object))

In [8]:
len(vasili_table.species.unique())

256

In [9]:
sub_vasili = vasili_table.loc[(vasili_table.finalQC != "fail")
                              & (vasili_table.cov_chrA >= 20)
                              & (vasili_table.remove_as_relative != True)
                             # & (vasili_table.mis_chrA < 0.01)
                              & (vasili_table.remove_manual != True)
                              & (~vasili_table.ID.str.startswith("SAMEA11633"))
                             ]
# Can get six more species if you use cov >= 10, but probably not worth the effort
len(sub_vasili.species.unique()), len(sub_vasili)

(240, 1390)

In [12]:
sub_vasili.species_genotyping.value_counts()

species_genotyping
Macaca_fascicularis_ssp        132
Macaca_mulatta_ssp             103
Papio_anubis_ssp                96
Pan_troglodytes_ssp             94
Papio_cynocephalus_ssp          70
                              ... 
Microcebus_griseorufus_ssp       1
Microcebus_jonahi_ssp            1
Microcebus_mittermeieri_ssp      1
Microcebus_tavaratra_ssp         1
Cercocebus_lunulatus_ssp         1
Name: count, Length: 251, dtype: int64

In [9]:
vasili_table.loc[(~vasili_table.species.isin(sub_vasili.species.unique())) & (vasili_table.cov_chrA >= 20)]

ID         ID_PGDP group           family         genus  \
199     SAMN11334319    SAMN11334319   STR   Cheirogaleidae  Cheirogaleus   
395          PD_0055         PD_0055   OWM  Cercopithecidae       Colobus   
473   SAMEA116330139  SAMEA116330139   STR        Lemuridae       Eulemur   
731   SAMEA116330175  SAMEA116330175   STR        Lemuridae     Hapalemur   
742          PD_0160         PD_0160   STR        Lemuridae     Hapalemur   
1593  SAMEA116330194  SAMEA116330194   STR    Lepilemuridae     Lepilemur   
2737         PD_0083         PD_0083   PLA      Pitheciidae      Pithecia   
2738         PD_0144         PD_0144   PLA      Pitheciidae      Pithecia   
2739         PD_0331         PD_0331   PLA      Pitheciidae      Pithecia   
2846  SAMEA116330132  SAMEA116330132   STR        Indriidae         Avahi   

             species subspecies                  species_full  \
199        crossleyi        ssp    Cheirogaleus_crossleyi_ssp   
395        polykomos        ssp         Colobus_polykomos_ssp   
473      cinereiceps        ssp       Eulemur_cinereiceps_ssp   
731           aureus        ssp          Hapalemur_aureus_ssp   
742     occidentalis        ssp    Hapalemur_occidentalis_ssp   
1593  sahamalazensis        ssp  Lepilemur_sahamalazensis_ssp   
2737      pissinatti        ssp       Pithecia_pissinatti_ssp   
2738      pissinatti        ssp       Pithecia_pissinatti_ssp   
2739      pissinatti        ssp       Pithecia_pissinatti_ssp   
2846    occidentalis        ssp        Avahi_occidentalis_ssp   

             species_short            species_genotyping  ...  \
199        C.crossleyi_ssp    Cheirogaleus_crossleyi_ssp  ...   
395        C.polykomos_ssp         Colobus_polykomos_ssp  ...   
473      E.cinereiceps_ssp       Eulemur_cinereiceps_ssp  ...   
731           H.aureus_ssp          Hapalemur_aureus_ssp  ...   
742     H.occidentalis_ssp    Hapalemur_occidentalis_ssp  ...   
1593  L.sahamalazensis_ssp  Lepilemur_sahamalazensis_ssp  ...   
2737      P.pissinatti_ssp       Pithecia_pissinatti_ssp  ...   
2738      P.pissinatti_ssp       Pithecia_pissinatti_ssp  ...   
2739      P.pissinatti_ssp       Pithecia_pissinatti_ssp  ...   
2846    A.occidentalis_ssp        Avahi_occidentalis_ssp  ...   

                             reason finalQC  Kuderna_AB_QC  Kuderna_COV_QC  \
199   wrong label; correct: C.major    fail            NaN             NaN   
395                             NaN    fail            NaN             NaN   
473                             NaN      ok            NaN             NaN   
731                             NaN    fail            NaN             NaN   
742                             NaN    fail           PASS            WARN   
1593                            NaN      ok            NaN             NaN   
2737                            NaN    fail            NaN             NaN   
2738                            NaN    fail            NaN             NaN   
2739                            NaN    fail            NaN             NaN   
2846                            NaN      ok            NaN             NaN   

      relatives  remove_as_relative  generation  founder  remove_manual  \
199         NaN                 NaN         NaN      NaN            NaN   
395         NaN                 NaN         NaN      NaN            NaN   
473         NaN                 NaN         NaN      NaN            NaN   
731         NaN                 NaN         NaN      NaN            NaN   
742         NaN                 NaN         NaN      NaN            NaN   
1593        NaN                 NaN         NaN      NaN            NaN   
2737        NaN                 NaN         NaN      NaN            NaN   
2738        NaN                 NaN         NaN      NaN            NaN   
2739        NaN                 NaN         NaN      NaN            NaN   
2846        NaN                 NaN         NaN      NaN            NaN   

      pedigree_source  
199               NaN  
395  

In [10]:
vasili_table.loc[vasili_table.species == "syrichta"]

ID  ID_PGDP group     family    genus   species subspecies  \
146  PD_0629  PD_0629   TAR  Tarsiidae  Carlito  syrichta        ssp   

             species_full   species_short    species_genotyping  ... reason  \
146  Carlito_syrichta_ssp  C.syrichta_ssp  Carlito_syrichta_ssp  ...    NaN   

    finalQC  Kuderna_AB_QC  Kuderna_COV_QC  relatives  remove_as_relative  \
146      ok           WARN            PASS        NaN                 NaN   

     generation  founder  remove_manual  pedigree_source  
146         NaN      NaN            NaN              NaN  

[1 rows x 93 columns]

In [11]:
count_sub = sub_vasili[["gSEX", "species"]].value_counts().reset_index()
len(count_sub.species.unique()), len(count_sub.loc[(count_sub.gSEX == "M") & (count_sub["count"] == 1)])

(240, 92)

In [12]:
len(count_sub.species.unique())

240

In [13]:
len(count_sub.loc[(count_sub.gSEX == "F") | ((count_sub.gSEX == "M") & (count_sub["count"] >= 2))].species.unique())

193

In [14]:
count_sub.loc[(count_sub.gSEX == "F")]

gSEX       species  count
0      F  fascicularis    104
1      F   troglodytes     55
2      F        anubis     54
3      F       mulatta     53
6      F       fuscata     42
..   ...           ...    ...
339    F       bicolor      1
340    F       hamlyni      1
341    F   brunnescens      1
342    F     avunculus      1
345    F      brunneus      1

[166 rows x 3 columns]

So after filtering, there are 166 species with at least one female in which you can determine chromosome X.
Adding samples with more than 1 male, we reach 195 (and assuming a present chrX). The last species only have a single male (or no identified females).

In [15]:
sub_vasili = vasili_table.loc[(vasili_table.finalQC != "fail")
                              & (vasili_table.cov_chrA >= 20)
                              & (vasili_table.remove_as_relative != True)
                              & (vasili_table.remove_manual != True)
                              & (~vasili_table.ID.str.startswith("SAMEA11633"))
                             ]

In [16]:
metadata_path = "/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/"
starting_dir = "/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/all_samples/bcf_step1/"

for species in sub_vasili.species_full.unique()[:2]:
    species_inds = sub_vasili.loc[sub_vasili.species_full == species]
    short_form = species.split("_")[0]
    regions_df = pd.read_csv(metadata_path+"{}_regions_and_batches.txt".format(short_form), sep="\t")
    reference = species_inds["reference"].unique()
    sample_names = ",".join(species_inds["ID"])
    all_chr_file = starting_dir.format(species)+species+"_all_chr.sorted.bcf"
    if not os.path.exists(all_chr_file):
        print("Filtered VCF does not exist ", species)
        continue
    if len(reference) > 1:
        print("Too many references ", species)
        continue
    zarr_input = []
    # Chromosome X
    zarr_input.extend(regions_df.loc[(regions_df.END >= 1000000) &
                                  (regions_df.FEMALE_PLOIDY == 2) &
                                  (regions_df.MALE_PLOIDY == 1) &
          (regions_df.REFERENCE_FOLDER == reference[0]+"_ssp")].CONTIG_ID.unique())

In [17]:
zarr_input

['NW_026988415.1',
 'NW_026988472.1',
 'NW_026988488.1',
 'NW_026988496.1',
 'NW_026988553.1']

In [18]:
reference, species

(array(['Aotus_nancymaae'], dtype=object), 'Aotus_azarai_ssp')

Some files fail dus to lack of information.

In [19]:
starting_dir = "/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/all_samples/bcf_step1/"
# starting_dir_gvcf = "/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/gVCF/"
metadata_path = "/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/" # Performing all selection with Vasili's stats
zarr_dir = "zarr_data_20x/"
vasili_stats = "~/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/plots/SupTable_Sample_Stats_wGT_QC.tsv"


vasili_table = pd.read_csv(vasili_stats, sep="\t")
sub_vasili = vasili_table.loc[(vasili_table.finalQC != "fail")
                              & (vasili_table.cov_chrA >= 20)
                              & (vasili_table.remove_as_relative != True)
                              & (vasili_table.remove_manual != True)
                              & (~vasili_table.ID.str.startswith("SAMEA11633"))
                             ]


In [20]:

for species in sub_vasili.species_full.unique()[:2]:
    species_inds = sub_vasili.loc[sub_vasili.species_full == species]
    short_form = species.split("_")[0]
    print(species, short_form)
    regions_df = pd.read_csv(metadata_path+"{}_regions_and_batches.txt".format(short_form), sep="\t")
    reference = species_inds["reference"].unique()
    sample_names = ",".join(species_inds["ID"])
    all_chr_file = starting_dir.format(species)+species+"_all_chr.sorted.bcf"
    if not os.path.exists(all_chr_file):
        print("Filtered VCF does not exist ", species)
        continue
    if len(reference) > 1:
        print("Too many references ", species)
        continue
    zarr_input = []

Allenopithecus_nigroviridis_ssp Allenopithecus
Aotus_azarai_ssp Aotus


In [21]:
sub_vasili.loc[sub_vasili.genus == "Gorilla"]#.species_full.unique()

ID       ID_PGDP group     family    genus   species  \
495  SAMEA2697037  SAMEA2697037   APE  Hominidae  Gorilla  beringei   
497  SAMEA2697039  SAMEA2697039   APE  Hominidae  Gorilla  beringei   
498  SAMEA3939556  SAMEA3939556   APE  Hominidae  Gorilla  beringei   
499  SAMEA3939557  SAMEA3939557   APE  Hominidae  Gorilla  beringei   
500  SAMEA3939558  SAMEA3939558   APE  Hominidae  Gorilla  beringei   
502  SAMEA3939560  SAMEA3939560   APE  Hominidae  Gorilla  beringei   
503  SAMEA3939561  SAMEA3939561   APE  Hominidae  Gorilla  beringei   
504  SAMEA3991715  SAMEA3991715   APE  Hominidae  Gorilla  beringei   
509  SAMEA2697040  SAMEA2697040   APE  Hominidae  Gorilla  beringei   
510  SAMEA2697041  SAMEA2697041   APE  Hominidae  Gorilla  beringei   
511  SAMEA2697042  SAMEA2697042   APE  Hominidae  Gorilla  beringei   
514  SAMN01920474  SAMN01920474   APE  Hominidae  Gorilla  beringei   
515  SAMN01920475  SAMN01920475   APE  Hominidae  Gorilla  beringei   
516       PD_0179       PD_0179   APE  Hominidae  Gorilla  beringei   
517  SAMEA1692350  SAMEA1692350   APE  Hominidae  Gorilla  beringei   
518  SAMEA1692351  SAMEA1692351   APE  Hominidae  Gorilla  beringei   
519  SAMEA1692352  SAMEA1692352   APE  Hominidae  Gorilla  beringei   
520  SAMEA1692353  SAMEA1692353   APE  Hominidae  Gorilla  beringei   
521  SAMEA1692354  SAMEA1692354   APE  Hominidae  Gorilla  beringei   
522  SAMEA1692355  SAMEA1692355   APE  Hominidae  Gorilla  beringei   
630  SAMEA5204233  SAMEA5204233   APE  Hominidae  Gorilla   gorilla   
631  SAMEA5204234  SAMEA5204234   APE  Hominidae  Gorilla   gorilla   
633  SAMEA5204236  SAMEA5204236   APE  Hominidae  Gorilla   gorilla   
635  SAMEA5204238  SAMEA5204238   APE  Hominidae  Gorilla   gorilla   
648  SAMN01920484  SAMN01920484   APE  Hominidae  Gorilla   gorilla   
649  SAMN01920485  SAMN01920485   APE  Hominidae  Gorilla   gorilla   
650  SAMN01920486  SAMN01920486   APE  Hominidae  Gorilla   gorilla   
651  SAMN01920487  SAMN01920487   APE  Hominidae  Gorilla   gorilla   
652  SAMN01920488  SAMN01920488   APE  Hominidae  Gorilla   gorilla   
675  SAMN12714420  SAMN12714420   APE  Hominidae  Gorilla   gorilla   
676  SAMN12714421  SAMN12714421   APE  Hominidae  Gorilla   gorilla   
677  SAMN35877944  SAMN35877944   APE  Hominidae  Gorilla   gorilla   
679       PD_0180       PD_0180   APE  Hominidae  Gorilla   gorilla   
682       PD_0183       PD_0183   APE  Hominidae  Gorilla   gorilla   
683       PD_0184       PD_0184   APE  Hominidae  Gorilla   gorilla   
685       PD_0186       PD_0186   APE  Hominidae  Gorilla   gorilla   
686       PD_0187       PD_0187   APE  Hominidae  Gorilla   gorilla   
687       PD_0188       PD_0188   APE  Hominidae  Gorilla   gorilla   
689       PD_0190       PD_0190   APE  Hominidae  Gorilla   gorilla   
690       PD_0191       PD_0191   APE  Hominidae  Gorilla   gorilla   
691       PD_0192       PD_0192   APE  Hominidae  Gorilla   gorilla   
692       PD_0193       PD_0193   APE  Hominidae  Gorilla   gorilla   
693       PD_0194       PD_0194   APE  Hominidae  Gorilla   gorilla   
694       PD_0195       PD_0195   APE  Hominidae  Gorilla   gorilla   
695       PD_0196       PD_0196   APE  Hominidae  Gorilla   gorilla   
696       PD_0197       PD_0197   APE  Hominidae  Gorilla   gorilla   
697       PD_0198       PD_0198   APE  Hominidae  Gorilla   gorilla   
698       PD_0260       PD_0260   APE  Hominidae  Gorilla   gorilla   
701  SAMEA2446072  SAMEA2446072   APE  Hominidae  Gorilla   gorilla   

    subspecies               species_full        species_short  \
495   beringei  Gorilla_beringei_beringei  G.beringei_beringei   
497   beringei  Gorilla_beringei_beringei  G.beringei_beringei   
498        ssp  Gorilla_beringei_beringei  G.beringei_beringei   
499        ssp  Gorilla_beringei_beringei  G.beringei_beringei   
500        ssp  Gorilla_beringei_beringei  G.beringei_beringei   
502        ssp  Gorilla_beringei_beringei  G.beringei_beringei   
50

In [22]:
regions_df.loc[regions_df.CONTIG_ID == "NW_026988493.1"]

REFERENCE_FOLDER          REGION       CONTIG_ID  START      END  \
81  Aotus_nancymaae_ssp  NW_026988493.1  NW_026988493.1      0  2145027   

    BATCH  FEMALE_PLOIDY  MALE_PLOIDY  
81     39              2            2

In [23]:
sub_vasili.species_full.unique()[:100]

array(['Allenopithecus_nigroviridis_ssp', 'Aotus_azarai_ssp',
       'Aotus_griseimembra_ssp', 'Aotus_nancymaae_ssp',
       'Aotus_trivirgatus_ssp', 'Aotus_vociferans_ssp',
       'Alouatta_belzebul_ssp', 'Alouatta_caraya_ssp',
       'Alouatta_discolor_ssp', 'Alouatta_juara_ssp',
       'Alouatta_macconnelli_ssp', 'Alouatta_palliata_ssp',
       'Alouatta_seniculus_puruensis', 'Alouatta_seniculus_ssp',
       'Ateles_belzebuth_ssp', 'Ateles_chamek_ssp',
       'Ateles_geoffroyi_ssp', 'Ateles_marginatus_ssp',
       'Ateles_paniscus_ssp', 'Brachyteles_hypoxanthus_ssp',
       'Lagothrix_lagotricha_ssp', 'Callimico_goeldii_ssp',
       'Callithrix_geoffroyi_ssp', 'Callithrix_jacchus_ssp',
       'Callithrix_kuhlii_ssp', 'Cebuella_niveiventris_ssp',
       'Cebuella_pygmaea_ssp', 'Mico_argentatus_ssp',
       'Mico_humeralifer_ssp', 'Mico_humilis_ssp', 'Carlito_syrichta_ssp',
       'Cephalopachus_bancanus_ssp', 'Tarsius_dentatus_ssp',
       'Tarsius_lariang_ssp', 'Tarsius_wallacei_ssp

In [24]:
pd.read_csv(metadata_path+"{}_regions_and_batches.txt".format("Gorilla"), sep="\t")

REFERENCE_FOLDER       REGION    CONTIG_ID     START        END  BATCH  \
0   Gorilla_gorilla_ssp  NC_073224.2  NC_073224.2         0  243847345      0   
1   Gorilla_gorilla_ssp  NC_086017.1  NC_086017.1         0  215253457      1   
2   Gorilla_gorilla_ssp  NC_073227.2  NC_073227.2         0  215373162      2   
3   Gorilla_gorilla_ssp  NC_073228.2  NC_073228.2         0  195332687      3   
4   Gorilla_gorilla_ssp  NC_073229.2  NC_073229.2         0  199452338      4   
5   Gorilla_gorilla_ssp  NC_073230.2  NC_073230.2         0  170803427      5   
6   Gorilla_gorilla_ssp  NC_073231.2  NC_073231.2         0  158533252      6   
7   Gorilla_gorilla_ssp  NC_073232.2  NC_073232.2         0  150804130      7   
8   Gorilla_gorilla_ssp  NC_073233.2  NC_073233.2         0  147876701      8   
9   Gorilla_gorilla_ssp  NC_073234.2  NC_073234.2         0  154228409      9   
10  Gorilla_gorilla_ssp  NC_073235.2  NC_073235.2         0  150248535     10   
11  Gorilla_gorilla_ssp  NC_073236.2  NC_073236.2         0  145906006     11   
12  Gorilla_gorilla_ssp  NC_073237.2  NC_073237.2         0  134695396     12   
13  Gorilla_gorilla_ssp  NC_073238.2  NC_073238.2         0  132018348     13   
14  Gorilla_gorilla_ssp  NC_073239.2  NC_073239.2         0  131692671     14   
15  Gorilla_gorilla_ssp  NC_073240.2  NC_073240.2         0  107915467     15   
16  Gorilla_gorilla_ssp  NC_073241.2  NC_073241.2         0  116054869     16   
17  Gorilla_gorilla_ssp  NC_073242.2  NC_073242.2         0  131347499     17   
18  Gorilla_gorilla_ssp  NC_073243.2  NC_073243.2         0  120452965     18   
19  Gorilla_gorilla_ssp  NC_073244.2  NC_073244.2         0   80312928     19   
20  Gorilla_gorilla_ssp  NC_073245.2  NC_073245.2         0   84443219     20   
21  Gorilla_gorilla_ssp  NC_073246.2  NC_073246.2         0   59507314     21   
22  Gorilla_gorilla_ssp  NC_086018.1  NC_086018.1         0   54769797     22   
23  Gorilla_gorilla_ssp      prePAR1  NC_073247.2         0    9544188     23   
24  Gorilla_gorilla_ssp         PAR1  NC_073247.2   9544188   12039748     23   
25  Gorilla_gorilla_ssp     postPAR1  NC_073247.2  12039748  177558554     23   
26  Gorilla_gorilla_ssp  NC_073248.2  NC_073248.2         0   67405748     24   
27  Gorilla_gorilla_ssp  NC_011120.1  NC_011120.1         0      16412     25   

    FEMALE_PLOIDY  MALE_PLOIDY  
0               2            2  
1               2            2  
2               2            2  
3               2            2  
4               2            2  
5               2            2  
6               2            2  
7               2            2  
8               2            2  
9               2            2  
10              2            2  
11              2            2  
12              2            2  
13              2            2  
14              2            2  
15              2            2  
16              2            2  
17              2            2  
18              2            2  
19              2            2  
20              2            2  
21              2            2  
22              2            2  
23              2            1  
24              2            2  
25              2            1  
26              0            1  
27              1            1

In [25]:
glob.glob("../zarr_20x_inds/Aotus_azarai_ssp/*/")

['../zarr_20x_inds/Aotus_azarai_ssp/NW_026988425.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988517.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988622.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988490.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988467.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988555.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988446.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988480.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988477.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988507.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988414.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988435.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988526.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988588.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988415.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988608.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988565.1/',
 '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988476.1/',
 '../zarr_20x_inds/Aotus_aza

Check of failed species for the first run.
They failed as the regions_df called them Macaca_thibetana_thibetana instead of Macaca_thibetana_ssp, for example. Fixed by using the correct metadata column.

In [26]:
vasili_table = pd.read_csv(vasili_stats, sep="\t")
sub_vasili = vasili_table.loc[(vasili_table.finalQC != "fail")
                              & (vasili_table.cov_chrA >= 20)
                              & (vasili_table.remove_as_relative != True)
                              & (vasili_table.remove_manual != True)
                              & (~vasili_table.ID.str.startswith("SAMEA11633"))
                             ]

failed_species = ['Saimiri_macrodon_ssp',
 'Macaca_thibetana_ssp',
 'Saimiri_sciureus_ssp',
 'Saimiri_oerstedii_ssp',
 'Saimiri_ustus_ssp',
 'Macaca_arctoides_ssp',
 'Macaca_leucogenys_ssp',
 'Macaca_assamensis_ssp',
 'Macaca_radiata_ssp',
 'Saimiri_cassiquiarensis_ssp',
 'Saimiri_boliviensis_ssp']


for species in failed_species[:2]:
    species_inds = sub_vasili.loc[sub_vasili.species_genotyping == species]
    print(species, species_inds)
    short_form = species.split("_")[0]
    regions_df = pd.read_csv(metadata_path+"{}_regions_and_batches.txt".format(short_form), sep="\t")
    reference = species_inds["reference"].unique()
    sample_names = ",".join(species_inds["ID"])
    all_chr_file = starting_dir.format(species)+species+"_all_chr.sorted.bcf"
    if not os.path.exists(all_chr_file):
        print("Filtered VCF does not exist ", species)
        continue
    if len(reference) > 1:
        print("Too many references ", species)
        continue
    zarr_input = []
    # Chromosome X
    zarr_input.extend(regions_df.loc[(regions_df.END >= 1000000) &
                                  (regions_df.FEMALE_PLOIDY == 2) &
                                  (regions_df.MALE_PLOIDY == 1) &
          (regions_df.REFERENCE_FOLDER == reference[0]+"_ssp")].CONTIG_ID.unique()[:])
    # Autosomes
    zarr_input.extend(regions_df.loc[(regions_df.END >= 1000000) &
                                  (regions_df.FEMALE_PLOIDY == 2) &
                                  (regions_df.MALE_PLOIDY == 2) &
                                  (~regions_df.CONTIG_ID.isin((zarr_input))) &
          (regions_df.REFERENCE_FOLDER == reference[0]+"_ssp")].CONTIG_ID.unique()[:])

Saimiri_macrodon_ssp            ID  ID_PGDP group   family    genus   species subspecies  \
3045  PD_0361  PD_0361   PLA  Cebidae  Saimiri  macrodon        ssp   
3046  PD_0362  PD_0362   PLA  Cebidae  Saimiri  macrodon        ssp   
3047  PD_0372  PD_0372   PLA  Cebidae  Saimiri  macrodon        ssp   

              species_full   species_short    species_genotyping  ... reason  \
3045  Saimiri_macrodon_ssp  S.macrodon_ssp  Saimiri_macrodon_ssp  ...    NaN   
3046  Saimiri_macrodon_ssp  S.macrodon_ssp  Saimiri_macrodon_ssp  ...    NaN   
3047  Saimiri_macrodon_ssp  S.macrodon_ssp  Saimiri_macrodon_ssp  ...    NaN   

     finalQC  Kuderna_AB_QC  Kuderna_COV_QC  relatives  remove_as_relative  \
3045      ok            NaN             NaN        NaN                 NaN   
3046      ok            NaN             NaN        NaN                 NaN   
3047      ok            NaN             NaN        NaN                 NaN   

      generation  founder  remove_manual  pedigree_source  


In [27]:
regions_df.loc[(regions_df.END >= 1000000) &
                                  (regions_df.FEMALE_PLOIDY == 2) &
                                  (regions_df.MALE_PLOIDY == 2) &
          (regions_df.REFERENCE_FOLDER == reference[0]+"_thibetana")].CONTIG_ID.unique()

array(['NC_065578.1', 'NC_065579.1', 'NC_065580.1', 'NC_065581.1',
       'NC_065582.1', 'NC_065583.1', 'NC_065584.1', 'NC_065585.1',
       'NC_065586.1', 'NC_065587.1', 'NC_065588.1', 'NC_065589.1',
       'NC_065590.1', 'NC_065591.1', 'NC_065592.1', 'NC_065593.1',
       'NC_065594.1', 'NC_065595.1', 'NC_065596.1', 'NC_065597.1'],
      dtype=object)

In [28]:
regions_df.loc[regions_df.REFERENCE_FOLDER.str.startswith(reference[0])]

REFERENCE_FOLDER          REGION       CONTIG_ID  START  \
3308  Macaca_thibetana_thibetana     NC_065578.1     NC_065578.1      0   
3309  Macaca_thibetana_thibetana     NC_065579.1     NC_065579.1      0   
3310  Macaca_thibetana_thibetana     NC_065580.1     NC_065580.1      0   
3311  Macaca_thibetana_thibetana     NC_065581.1     NC_065581.1      0   
3312  Macaca_thibetana_thibetana     NC_065582.1     NC_065582.1      0   
...                          ...             ...             ...    ...   
3904  Macaca_thibetana_thibetana  NW_026089386.1  NW_026089386.1      0   
3905  Macaca_thibetana_thibetana  NW_026089387.1  NW_026089387.1      0   
3906  Macaca_thibetana_thibetana  NW_026089388.1  NW_026089388.1      0   
3907  Macaca_thibetana_thibetana  NW_026089389.1  NW_026089389.1      0   
3908  Macaca_thibetana_thibetana  NW_026089390.1  NW_026089390.1      0   

            END  BATCH  FEMALE_PLOIDY  MALE_PLOIDY  
3308  220358725      0              2            2  
3309  194429148      1              2            2  
3310  185122172      2              2            2  
3311  168803367      3              2            2  
3312  185943000      4              2            2  
...         ...    ...            ...          ...  
3904       5000     21              2            2  
3905       5000     21              2            2  
3906     161000     21              2            2  
3907     150144     21              2            2  
3908      10000     21              2            2  

[601 rows x 8 columns]

In [29]:
regions_df

REFERENCE_FOLDER          REGION       CONTIG_ID  START  \
0        Macaca_fascicularis_ssp      CP141361.1      CP141361.1      0   
1        Macaca_fascicularis_ssp      CP141360.1      CP141360.1      0   
2        Macaca_fascicularis_ssp      CP141359.1      CP141359.1      0   
3        Macaca_fascicularis_ssp      CP141358.1      CP141358.1      0   
4        Macaca_fascicularis_ssp      CP141357.1      CP141357.1      0   
...                          ...             ...             ...    ...   
3904  Macaca_thibetana_thibetana  NW_026089386.1  NW_026089386.1      0   
3905  Macaca_thibetana_thibetana  NW_026089387.1  NW_026089387.1      0   
3906  Macaca_thibetana_thibetana  NW_026089388.1  NW_026089388.1      0   
3907  Macaca_thibetana_thibetana  NW_026089389.1  NW_026089389.1      0   
3908  Macaca_thibetana_thibetana  NW_026089390.1  NW_026089390.1      0   

            END  BATCH  FEMALE_PLOIDY  MALE_PLOIDY  
0     234122563      0              2            2  
1     203129947      1              2            2  
2     200656507      2              2            2  
3     173030664      3              2            2  
4     194799049      4              2            2  
...         ...    ...            ...          ...  
3904       5000     21              2            2  
3905       5000     21              2            2  
3906     161000     21              2            2  
3907     150144     21              2            2  
3908      10000     21              2            2  

[3909 rows x 8 columns]